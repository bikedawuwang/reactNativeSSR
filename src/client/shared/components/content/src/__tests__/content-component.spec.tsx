import * as React from 'react';
import { View, Text } from 'react-native-web';
import { shallow } from 'enzyme';
// Web
import SharedComponentWeb from '../index';

// Common
import ChildComponent from '../render-app';

describe('SharedComponent <SharedComponent />', () => {
	it('Shared Component Web renders correctly', () => {
		const sharedComponentWeb = shallow(<SharedComponentWeb />);

		// Interaction demo
		expect(sharedComponentWeb.text()).toEqual('<App />');

		// Snapshot demo
		expect(shallow).toMatchSnapshot();
	});
	it('Shared Component renders child component correctly', () => {
		const childComponent = shallow(<ChildComponent api={{ View, Text }} />);

		// Interaction demo
		expect(childComponent.text()).toEqual('<View />');

		// Snapshot demo
		expect(shallow).toMatchSnapshot();
	});
});
