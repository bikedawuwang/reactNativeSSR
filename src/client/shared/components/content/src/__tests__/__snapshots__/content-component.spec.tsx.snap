// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedComponent <SharedComponent /> Shared Component Web renders correctly 1`] = `[Function]`;

exports[`SharedComponent <SharedComponent /> Shared Component renders child component correctly 1`] = `[Function]`;
